ballerina-to-openapi/src/main/java/io/ballerina/openapi/service/mapper/ResourceMapperImpl.java

```diff
@@ -72,9 +80,10 @@ public class ResourceMapperImpl implements ResourceMapper {
         this.resources = resources;
         this.additionalData = additionalData;
         this.treatNilableAsOptional = treatNilableAsOptional;
+        this.hateoasMapper = new HateoasMapper();
     }
 
-    public void setOperation() {
+    public void setOperation(String packageId, SemanticModel semanticModel, ServiceDeclarationNode service) {
```

Comment:
We can get the semanticModel from the `additionalData`

Reply:
Thanks for the comment! This is resolved now!